workflow/controller/operator.go

```diff
@@ -2366,7 +2366,9 @@ func (woc *wfOperationCtx) markWorkflowPhase(ctx context.Context, phase wfv1.Wor
 			}
 			woc.updated = true
 		}
-		woc.controller.queuePodForCleanup(woc.wf.Namespace, woc.getAgentPodName(), deletePod)
+		if woc.hasPluginNodes() {
```

Comment:
The agent isn't just for plugin nodes, it's also for other built-in template types like `http` etc as I mentioned in https://github.com/argoproj/argo-workflows/issues/12659#issuecomment-1939189088

Reply:
@alexec can u confirm what node types have agent pod? i saw https://github.com/argoproj/argo-workflows/pull/7256 introduced this delete agent pod line and that was the same PR that introduced plugin node

Reply:
Follow-up thread [below](https://github.com/argoproj/argo-workflows/pull/12679#discussion_r1495400802). Eventually resolved in #13294